### PR TITLE
fix: Fixes issues active page update and deleting all rows in a page

### PR DESCRIPTION
**Pull request by zeeklop**
Fix small bugs on `PaginatedList.tsx`

## Description

- Update internal `currentPage` when `activePage` changes.
- Change to immediate previous page when the current page doesn't have any items to show.
- Fix issue when changing page size and the number of pages would stay the same. 

## Motivation and Context

While working with the component `PaginatedList` I noticed a few inconsistencies when parameters would change from the parent component. This small fix is to make this component to act in a more predictable way.

## How Has This Been Tested?

I've tested the use of this component manually and with unit tests.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/common/PaginatedList.tsx [hunk 1]
```diff
@@ -60,38 +59,39 @@ const PaginatedList = ({
...
+  const [{ currentPage, currentPageSize }, setPagination] = React.useState({
+    currentPage: useQueryParameter ? page : Math.max(activePage, INITIAL_PAGE),
+    currentPageSize: (useQueryParameter && showPageSizeSelect) ? queryParamPageSize : propPageSize,
+  });
+
+  const numberPages = React.useMemo(() => (
+    currentPageSize > 0 ? Math.ceil(totalItems / currentPageSize) : 0
+  ), [currentPageSize, totalItems]);
 
   const _onChangePageSize = (event: React.ChangeEvent<HTMLOptionElement>) => {
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/common/PaginatedList.tsx:72]: Let's use `useCallback` here as well?

**Review by linuspahl — CHANGES_REQUESTED** [review]
I found a problem which also exists in the previous version, but maybe we can fix it in this PR?

When you change the values for `DEFAULT_PAGE_SIZES`, for example from `[10, 50, 100]` to `[5, 10, 50, 100]`, it selects 5 as the page size by default, but it still lists more items. I tested it with the dashboards list.

File: graylog2-web-interface/src/components/common/PaginatedList.tsx (hunk 1 above)
Comment (zeeklop) [reply graylog2-web-interface/src/components/common/PaginatedList.tsx:72]: sounds good. I'll take a look

**Comment by zeeklop** [comment]
> 

Sure thing. I think I have an idea why this is happening. I'll take a look. Thanks

**Comment by zeeklop** [comment]
> I found a problem which also exists in the previous version, but maybe we can fix it in this PR?
> 
> When you change the values for `DEFAULT_PAGE_SIZES`, for example from `[10, 50, 100]` to `[5, 10, 50, 100]`, it selects 5 as the page size by default, but it still lists more items. I tested it with the dashboards list.

@linuspahl 
After further analysis, I found out the issue is with the hook implementation. I tried a few different solutions for this issue. The easiest one is to not check for the existence of the page size inside the hook. Without this check, users would be able to set the parameter to any number they want, which in some cases could crash the application.

In my opinion, this issue, should be addressed in a different PR since it is not related to the paginated list component. I can open a new issue to work on it.

**Review by linuspahl — APPROVED** [review]
Looks good to me.

**Pull request merged**